catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/impl/CachedResourceMetacardComparatorTest.java

```diff
@@ -0,0 +1,299 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.cache.impl;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static ddf.catalog.cache.impl.CachedResourceMetacardComparator.isSame;
+
+import java.net.URI;
+import java.net.URL;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.BasicTypes;
+import ddf.catalog.data.impl.EmptyMetacardType;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.MetacardTypeImpl;
+
+public class CachedResourceMetacardComparatorTest {
+
+    private MetacardImpl cachedMetacard;
+
+    private MetacardImpl updatedMetacard;
+
+    @Before
+    public void setup() throws Exception {
+        Instant createdDate = Instant.now();
+        Instant effectiveDate = createdDate.plus(1, ChronoUnit.HOURS);
+        Instant modificationDate = createdDate.plus(2, ChronoUnit.DAYS);
+        Instant expirationDate = createdDate.plus(60, ChronoUnit.DAYS);
+
+        String metacardId = UUID.randomUUID()
+                .toString();
+
+        cachedMetacard = createMetacard(metacardId,
+                createdDate,
+                effectiveDate,
+                expirationDate,
+                modificationDate);
+        updatedMetacard = createMetacard(metacardId,
+                createdDate,
+                effectiveDate,
+                expirationDate,
+                modificationDate);
+    }
+
+    @Test
+    public void isSameMetacard() throws Exception {
+        assertThat(isSame(cachedMetacard, updatedMetacard), is(true));
```

Comment:
No point to change, but these sure would be nice and concise with spock :-)
(if i'm feeling rambunctious maybe i'll give my hand a try at seeing what it would look like)